src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -429,38 +430,70 @@ protected String addACHAccount(Person payee, PayeeACHAccountExtractDetail achDet
-        LOG.warn("updateACHAccountIfNecessary: " + processingResults.toString());
+        
         return processingResults.toString();
     }
 
+    protected boolean accountHasChanged(PayeeACHAccountExtractDetail achDetail, PayeeACHAccount achAccount) {
+        return !StringUtils.equals(achDetail.getBankRoutingNumber(), achAccount.getBankRoutingNumber())
+                || !StringUtils.equals(achDetail.getBankAccountNumber(), achAccount.getBankAccountNumber())
+                || !StringUtils.equals(
+                        getACHTransactionCode(achDetail.getBankAccountType()), achAccount.getBankAccountTypeCode());
+    }
+
+    protected String updateACHAccount(Person payee, PayeeACHAccountExtractDetail achDetail, PayeeACHAccount achAccount) {
+        PayeeACHAccount oldAccount = achAccount;
+        PayeeACHAccount newAccount = (PayeeACHAccount) ObjectUtils.deepCopy(oldAccount);
+        
+        newAccount.setBankRoutingNumber(achDetail.getBankRoutingNumber());
+        newAccount.setBankAccountNumber(achDetail.getBankAccountNumber());
+        newAccount.setBankAccountTypeCode(getACHTransactionCode(achDetail.getBankAccountType()));
+        
+        return createAndRouteDocumentForACHAccountUpdate(payee, achDetail, oldAccount, newAccount);
+    }
+
+    protected String createAndRouteDocumentForACHAccountUpdate(
+            Person payee, PayeeACHAccountExtractDetail achDetail, PayeeACHAccount oldAccount, PayeeACHAccount newAccount) {
+        try {
+            MaintenanceDocument paatDocument = (MaintenanceDocument) documentService.getNewDocument(CUPdpConstants.PAYEE_ACH_ACCOUNT_EXTRACT_MAINT_DOC_TYPE);
+            paatDocument.getDocumentHeader().setDocumentDescription(getDocumentDescription(
+                    payee, PayeeIdTypeCodes.ENTITY.equals(newAccount.getPayeeIdentifierTypeCode()), false));
+            
+            paatDocument.getOldMaintainableObject().setDataObject(oldAccount);
+            paatDocument.getNewMaintainableObject().setDataObject(newAccount);
+            paatDocument.getNewMaintainableObject().setMaintenanceAction(KFSConstants.MAINTENANCE_EDIT_ACTION);
+            
+            addNote(paatDocument, parameterService.getParameterValueAsString(
+                    PayeeACHAccountExtractStep.class, CUPdpParameterConstants.GENERATED_PAYEE_ACH_ACCOUNT_DOC_NOTE_TEXT));
+
+            paatDocument = (MaintenanceDocument) documentService.routeDocument(
+                    paatDocument, CUPdpConstants.PAYEE_ACH_ACCOUNT_EXTRACT_ROUTE_ANNOTATION, null);
+            sendPayeeACHAccountAddOrUpdateEmail((PayeeACHAccount) paatDocument.getNewMaintainableObject().getDataObject(), payee,
+                    getEmailSubjectForUpdatedPayeeACHAccount(), getUnresolvedEmailBodyForUpdatedPayeeACHAccount());
+        } catch (Exception e) {
+            LOG.error("createAndRouteDocumentForACHAccountUpdate STE " + e.getStackTrace() + e.toString());
+            LOG.error(getFailRequestMessage(e));

```

Comment:
this looks like we are logging the exception twice, can the second log statement be eliminated by just adding "e" to the first?

Reply:
Ok, I'll work on cleaning up this logging.